addons/stock/models/stock_quant.py

```diff
@@ -378,6 +468,46 @@ def _onchange_serial_number(self):
             if message:
                 return {'warning': {'title': _('Warning'), 'message': message}}
 
+    def _apply_inventory(self):
+        moves = OrderedSet()
+        for quant in self:
+            # Create and validate a move so that the quant matches its `inventory_quantity`.
+            if float_compare(quant.inventory_diff_quantity, 0, precision_rounding=quant.product_uom_id.rounding) > 0:
+                move_vals = quant._get_inventory_move_values(quant.inventory_diff_quantity, quant.product_id.with_company(
+                    quant.company_id).property_stock_inventory, quant.location_id)
+            else:
+                move_vals = quant._get_inventory_move_values(-quant.inventory_diff_quantity, quant.location_id, quant.product_id.with_company(
+                    quant.company_id).property_stock_inventory, out=True)
+            moves.add(quant.env['stock.move'].with_context(inventory_mode=False).create(move_vals).id)
```

Comment:
I don't get this context on stock move. I though it was specific to quant. 
Also you should create stock moves in batch

Reply:
Batch create fix done. As for the context, it's because when `moves._action_done` is called, it creates/writes quants and we need to make sure we're not in `inventory_mode` when that happens (otherwise some issues might pop up).